### ISSUE TITLE
Cannot use vars for chef provisioner bool arguments

### ISSUE THREAD

**Issue by thegedge** [hashicorp/terraform#2872]
```sh
variable "skip_chef_install" {
  default = true
}

resource "aws_instance" "foo" {
  ...
  provisioner "chef" {
    ...
    skip_install = "${var.skip_chef_install}"
    ...
  }
  ...
}
```

```console
$ terraform plan --module-depth=-1
There are warnings and/or errors related to your configuration. Please
fix these before continuing.

Errors:

  * aws_instance.foo: 1 error(s) decoding:

* cannot parse 'skip_install' as bool: strconv.ParseBool: parsing "${var.skip_chef_install}": invalid syntax
```

**Comment by svanharmelen** [comment]
@thegedge thanks for reporting this one! This is an interesting one, so I'll have a closer look when I have a little more time on my hands...

Will get back to you when I have a solution or course!

**Comment by svanharmelen** [comment]
@thegedge it took a little while to find some time to work on this, but I managed to reproduce and fix it today in PR #3049

**Comment by thegedge** [comment]
Ah, great. Thanks for the fix, @svanharmelen. Next time I'll have an idea of what to look for to send in a PR myself :)
